compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -571,8 +571,8 @@ class Namer { typer: Typer =>
 
     /** Create links between companion object and companion class */
     def createLinks(classTree: TypeDef, moduleTree: TypeDef)(implicit ctx: Context) = {
-      val claz = ctx.effectiveScope.lookup(classTree.name.encode)
-      val modl = ctx.effectiveScope.lookup(moduleTree.name.encode)
+      val claz = ctx.denotNamed(classTree.name).symbol
```

Comment:
Why does `denotNamed` makes sense here? I changed it to `effectiveScope.lookup` in https://github.com/lampepfl/dotty/commit/b6411818b2e84c0a28c501e2814de4cfde720208 assuming it was the right thing to do. Doesn't using `denotNamed` means that we could find a companion in an outer scope which would be very wrong?